lib/pry-doc/pry_ext/show_source_with_c_internals/c_extractor.rb

```diff
@@ -0,0 +1,195 @@
+require 'fileutils'
+require_relative 'c_file'
+
+class CExtractor
+  include Pry::Helpers::Text
+
+  class << self
+    attr_accessor :file_cache
+  end
+  @file_cache = {}
+
+  attr_reader :opts
+
+  def initialize(opts)
+    @opts = opts
+  end
+
+  def balanced?(str)
+    tokens = CodeRay.scan(str, :c).tokens.each_slice(2).to_a
+    tokens.count { |v|
+      v.first =~ /{/ && v.last == :operator } == tokens.count { |v|
+      v.first =~ /}/ && v.last == :operator
+    }
```

Comment:
Is it possible to reformat this? It's a little hard to read. It took me a while to spot `==`.